### PR TITLE
Rabin Karp in C

### PR DESCRIPTION
**Pull request by disha2sinha**
Fixes #2161 

#### Checklist:
- [X] 4 space indentation.
- [X] Coding conventions are followed.
- [X] Input is taken dynamically.
- [X] Sample Input / Output is added at the end of file.
- [X] Logic Documentation (Comments).
- [X] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Brief description added
- Variables added

#### Languages Used:
- C

#### Files Added:
- Rabin_Karp.c

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Rabin_Karp/Rabin_Karp.c [hunk 1]
```diff
@@ -0,0 +1,72 @@
+#include <stdio.h>
```
Comment (singh-shreya6) [inline Rabin_Karp/Rabin_Karp.c:1]: Please add a brief problem description. 1-2 lines

File: Rabin_Karp/Rabin_Karp.c [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    for (int i = 0; i < pattern_len - 1; i++)
+    {
+        hash = (hash * 256) % 149;
+        // there are 256 total characters in Ascii table & 149 is a prime number chosen randomly
+    }
+    int hash_txt = 0, hash_pat = 0;
+
+    for (int i = 0; i < pattern_len; i++)
+    {
+        hash_txt = (hash_txt * 256 + text [i]) % 149; // hash value for text
```
Comment (singh-shreya6) [inline Rabin_Karp/Rabin_Karp.c:19]: Can you please store the values 256 and 149 into some variable and use it.

**Comment by singh-shreya6** [comment]
Also complete the PR template

**Comment by disha2sinha** [comment]
Changes made

File: Rabin_Karp/Rabin_Karp.c [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+value of current substring of text and if the hash values match then only it starts matching individual characters*/
+#include <stdio.h>
+#include <string.h>
+
+void Rabin_Karp (char *text, char *pattern)
+{
+    int text_len = strlen (text);
+    int pattern_len = strlen (pattern);
+    int hash = 1;
+    int asciichars=256;
```
Comment (singh-shreya6) [inline Rabin_Karp/Rabin_Karp.c:11]: space around =

File: Rabin_Karp/Rabin_Karp.c [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    int prime=149;
```
Comment (singh-shreya6) [inline Rabin_Karp/Rabin_Karp.c:12]: space around =

File: Rabin_Karp/Rabin_Karp.c [hunk 5]
```diff
@@ -0,0 +1,76 @@
+/*Rabin Karp algorithm is a string matching algorithm which matches the hash value of the pattern with the hash 
+value of current substring of text and if the hash values match then only it starts matching individual characters*/
+#include <stdio.h>
```
Comment (singh-shreya6) [inline Rabin_Karp/Rabin_Karp.c:3]: Please add a blank line before this.

**Pull request merged**